src/System.IO.Pipelines.Networking.Tls/Internal/Sspi/Interop.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+
+namespace System.IO.Pipelines.Networking.Tls.Internal.Sspi
+{
+    internal unsafe static class Interop
+    {
+        internal const string Dll = "sspicli.dll";
+
+        [DllImport(Dll, ExactSpelling = true, SetLastError = true)]
+        internal static extern int EnumerateSecurityPackagesW([Out] out int pkgnum, [Out] out SecPkgInfo* info);
+
+        [DllImport(Dll, ExactSpelling = true, SetLastError = true)]
+        internal static extern int FreeContextBuffer([In] IntPtr contextBuffer);
+
+        [DllImport(Dll, ExactSpelling = true, CharSet = CharSet.Unicode, SetLastError = true)]
+        internal unsafe static extern SecurityStatus AcquireCredentialsHandleW([In] string principal, [In] string moduleName, [In] int usage, [In] void* logonID, [In] ref SecureCredential authdata, [In] void* keyCallback, [In] void* keyArgument, ref SSPIHandle handlePtr, [Out] out long timeStamp);
+
+        [DllImport(Dll, ExactSpelling = true, SetLastError = true)]
+        internal static extern int DeleteSecurityContext(ref SSPIHandle handlePtr);
+
+        [DllImport(Dll, ExactSpelling = true, SetLastError = true)]
+        internal static extern int FreeCredentialsHandle(ref SSPIHandle handlePtr);
+
+        [DllImport(Dll, ExactSpelling = true, SetLastError = true)]
+        internal unsafe static extern SecurityStatus AcceptSecurityContext(ref SSPIHandle credentialHandle, [In] void* inContextPtr, SecurityBufferDescriptor* inputBuffer, [In] ContextFlags inFlags, [In] Endianness endianness, [In, Out] void* newContextPtr, SecurityBufferDescriptor* outputBuffer, [In, Out] ref ContextFlags attributes, out long timeStamp);
+
+        internal static CipherInfo GetCipherInfo(SSPIHandle _contextPointer)
+        {
+            _SecPkgContext_ConnectionInfo connInfo;
+            QueryContextAttributesW(ref _contextPointer, ContextAttribute.ConnectionInfo, out connInfo);
```

Comment:
I would actually say, this needs improvement all round. This was added at the last minute before benchmarks to make sure that the ciphers that SslStream and this lib were the same to compare apples to apples. It needs a proper API to get more detail